src/gui/vectortile/qgsvectortilebasiclabelingwidget.cpp

```diff
@@ -0,0 +1,458 @@
+/***************************************************************************
+  qgsvectortilebasiclabelingwidget.cpp
+  --------------------------------------
+  Date                 : May 2020
+  Copyright            : (C) 2020 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsvectortilebasiclabelingwidget.h"
+
+#include "qgsvectortilebasiclabeling.h"
+#include "qgsvectortilelayer.h"
+
+#include "qgslabelinggui.h"
+
+#include <QMenu>
+
+///@cond PRIVATE
+
+
+class QgsVectorTileBasicLabelingListModel : public QAbstractListModel
+{
+  public:
```

Comment:
Q_OBJECT

Reply:
Just wondering - why add it if it is not needed here? It's a private class, there are no signals/slots, there's no qobject_cast in use...

Reply:
I've looked into this before, and advice from qt (and kdab) is that it's required regardless of whether the subclass signals or slots.

Reply:
Okay cool - fixed in 2cd15ea - I see it in the Qt docs now: "We strongly recommend the use of this macro in all subclasses of QObject regardless of whether or not they actually use signals, slots and properties, since failure to do so may lead certain functions to exhibit strange behavior."